### PR TITLE
Iss790 rtpc disentangler

### LINKED ISSUE #790: Improve RTPC track disentagler

### PR DESCRIPTION
**Pull request by Hattawy**
final updated rtpc code including updates on the disentangler and updated the helix fit code to fit the first part of the back-bend tracks

### CODE REVIEW

File: reconstruction/rtpc/src/main/java/org/jlab/rec/rtpc/hit/TrackDisentangler.java [hunk 1]
```diff
@@ -22,85 +26,181 @@ public class TrackDisentangler {
     private double maxdeltaphi = 0.10;
     private double maxdeltazgap = 10;
     private double maxdeltaphigap = 0.12;
-    
-    public TrackDisentangler(HitParameters params, boolean disentangle){       
+    private double c0 = 20;
+    private double c1 = 0.07;
+    private double c2 = 8;
+    private double c3 = 240;
+    private int eventnum = 777;
```
Comment (raffaelladevita) [inline reconstruction/rtpc/src/main/java/org/jlab/rec/rtpc/hit/TrackDisentangler.java]: it seems this eventnum and the corresponding variable in the constructor are leftover from some debugging, right? can they be removed?

File: reconstruction/rtpc/src/main/java/org/jlab/service/rtpc/RTPCEngine.java [hunk 2]
```diff
@@ -102,13 +101,15 @@ public boolean processDataEvent(DataEvent event) {
             return true;
         }
 
-         int runNo = 10;
+        int runNo = 10;
+        int eventNo = 777;
```
Comment (raffaelladevita) [inline reconstruction/rtpc/src/main/java/org/jlab/service/rtpc/RTPCEngine.java:106]: same as above

File: reconstruction/rtpc/src/main/java/org/jlab/rec/rtpc/hit/TrackDisentangler.java (hunk 1 above)
Comment (Hattawy) [reply reconstruction/rtpc/src/main/java/org/jlab/rec/rtpc/hit/TrackDisentangler.java]: c0, c1, c2, and c3 must be defined here. I removed the initialization for them as they have to be reading the constants from ccdb. Regarding,      private int eventnum = 777; 
It has been removed as it got deined in the RTPC engine earlier.

File: reconstruction/rtpc/src/main/java/org/jlab/service/rtpc/RTPCEngine.java (hunk 2 above)
Comment (Hattawy) [reply reconstruction/rtpc/src/main/java/org/jlab/service/rtpc/RTPCEngine.java:106]: they must be initialized here. Removing the definition and the initialization would crashing building coatjava. 

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0:compile (default-compile) on project clas12detector-rtpc: Compilation failure: Compilation failure: 
[ERROR] /w/hallb-scshelf2102/clas12/mhattawy/clara-bonusdev/clas12-offline-software/reconstruction/rtpc/src/main/java/org/jlab/service/rtpc/RTPCEngine.java:[121,77] variable runNo might not have been initialized
[ERROR] /w/hallb-scshelf2102/clas12/mhattawy/clara-bonusdev/clas12-offline-software/reconstruction/rtpc/src/main/java/org/jlab/service/rtpc/RTPCEngine.java:[136,77] variable eventNo might not have been initialized

**Comment by Hattawy** [comment]
[https://clasweb.jlab.org/wiki/images/5/53/RG-F_reconstruction_improvments.pdf](url)
slides presented during the clas12 software meeting on May 19th, 2022.

File: reconstruction/rtpc/src/main/java/org/jlab/service/rtpc/RTPCEngine.java (hunk 2 above)
Comment (raffaelladevita) [reply reconstruction/rtpc/src/main/java/org/jlab/service/rtpc/RTPCEngine.java:106]: Taken care of in commit https://github.com/JeffersonLab/clas12-offline-software/pull/833/commits/729a5f5afbe86c4c8943fafd63304f48d8d7e89c

**Review by raffaelladevita — APPROVED** [review]

**Pull request merged**